### PR TITLE
Admin Router: Finalize tests for Admin Router

### PR DESCRIPTION
**Pull request by vespian**
## High Level Description

This PR:
* add missing tests for all AR endpoints
* reworks generic tests in order to simplify configuration
* adds "is unauth(n|z) access possible" generic tests
* adds "response is correct" generic tests (e.g. whether caching headers are properly set)
* make `generic_correct_upstream_request_test` more strict
* adds missing tests for:
  * DCOS_OSS-1005 
  * DCOS-15510
  * DCOS-16021
* refactors generalised tests into a library so that we can walk around the problem of fixtures not being included in directories above `ee/` and `open/` 
* fix bug in X-Accell-Buffering passing for Open DC/OS Agents
* force Nginx to honour logging settings in `init_by_lua` stage
* make `/marathon` a redirect-only endpoint

## Related Issues

- https://jira.mesosphere.com/browse/DCOS_OSS-1005 Admin Router: add tests that cover the changes introduced via DCOS_OSS-1003
- https://jira.mesosphere.com/browse/DCOS-15362 Admin Router: Tests of the nginx.(agent|master).conf
- https://jira.mesosphere.com/browse/DCOS-15510 Admin Router: add test for dcos-version endpoint in agents.
- https://jira.mesosphere.com/browse/DCOS-16021 Admin Router: add test for DCOS-13976: Disable opening the UI in iframe
- https://jira.mesosphere.com/browse/DCOS_OSS-1163 "mycluster.com/marathon" routes to "mycluster.com/ui" => 404

## Dependencies
EE PR: https://github.com/mesosphere/dcos-enterprise/pull/979

### CODE REVIEW

**Comment by vespian** [comment]
@mesosphere-mergebot label Ready for review

File: packages/adminrouter/extra/src/nginx.agent.conf [hunk 1]
```diff
@@ -4,6 +4,10 @@ include common/main.conf;
 http {
     resolver 198.51.100.1:53 198.51.100.2:53 198.51.100.3:53 valid=60s;
 
+	# Check
+	# https://github.com/openresty/lua-nginx-module/issues/467#issuecomment-305529857
```
Comment (mhrabovcin) [inline packages/adminrouter/extra/src/nginx.agent.conf]: 😮

File: packages/adminrouter/extra/src/test-harness/modules/generic_test_code/common.py [hunk 2]
```diff
@@ -63,6 +64,38 @@ def generic_no_slash_redirect_test(ar, path):
     assert r.headers['Location'] == url + '/'
 
 
+def generic_response_headers_verify_test(
+        ar, auth_header, path, assert_headers=None, assert_headers_absent=None):
+    """Test if response sent by AR is correct
```
Comment (mhrabovcin) [inline packages/adminrouter/extra/src/test-harness/modules/generic_test_code/common.py]: Can you expand what does `correct` mean?

File: packages/adminrouter/extra/src/test-harness/modules/runner/open.py [hunk 3]
```diff
@@ -31,6 +31,6 @@ def __init__(self,
         """
         NginxBase.__init__(self, **base_kwargs)
 
-        self._set_ar_env_from_val("OUATH_CLIENT_ID", ouath_client_id)
-        self._set_ar_env_from_val("OUATH_AUTH_REDIRECTOR", ouath_auth_redirector)
+        self._set_ar_env_from_val("OAUTH_CLIENT_ID", ouath_client_id)
```
Comment (mhrabovcin) [inline packages/adminrouter/extra/src/test-harness/modules/runner/open.py]: 👍

**Comment by vespian** [comment]
Closed in favour of https://github.com/dcos/dcos/pull/1611

**Pull request closed without merging**